packages/ui/src/stories/molecules/CouponInput.stories.tsx

```diff
@@ -0,0 +1,47 @@
+import type { Meta, StoryObj } from '@storybook/react';
+import React from 'react';
+
+import { CouponInput } from '../../components';
+
+const meta: Meta<typeof CouponInput> = {
+  component: CouponInput,
+  render: args => {
+    const [code, setCode] = React.useState('');
+
+    const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setCode(event.target.value);
+    };
+
+    return (
+      <CouponInput
+        {...args}
+        code={code}
+        onChange={onChange}
+        onApply={() => console.log('Coupon applied')}
+        onDelete={() => console.log('Coupon deleted')}
```

Comment:
Implement the `onApply` and `onDelete` function so that the button works on storybook.